### PR TITLE
fix all issues reported by the default golangci linters

### PR DESCRIPTION
**Pull request by tariq1890**
This fixes all the issues reported by the default linter tools used by golangci.

### CODE REVIEW

File: model/silence_test.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func TestMatcherValidate(t *testing.T) {
 			t.Errorf("%d. Expected error %q but got none", i, c.err)
 			continue
 		}
-		if c.err == "" && err != nil {
```
Comment (tariq1890) [inline model/silence_test.go]: This is a redundant tautology.

File: config/tls_config_test.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func TestValidTLSConfig(t *testing.T) {
 		}
 		// non-nil functions are never equal.
 		got.GetClientCertificate = nil
-		if !reflect.DeepEqual(*got, *cfg.config) {
+		if !reflect.DeepEqual(got, cfg.config) {
```
Comment (tariq1890) [inline config/tls_config_test.go]: We shouldn't compare actual structs which contain mutexes. Compare using the pointers instead.

**Comment by tariq1890** [comment]
@brian-brazil @simonpasquier FYI

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**